packages/core/crates/core-misc/src/heartbeat.rs

```diff
@@ -0,0 +1,56 @@
+use blake2::{Blake2s256, Digest};
+
+/// Takes a ping request message and returns a ping response message
+pub fn generate_ping_response(req: Vec<u8>) -> Vec<u8> {
```

Comment:
Let's use `&[u8]` instead as type of the functions argument. You're not using any Vector's features, just calling `.as_slice()` on it straightaway.

As return type I'd use `Box<[u8]>` because since it is a hash, you won't be resizing it/adding/removing any elements from it.